src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py

```diff
@@ -270,8 +270,11 @@ def in_expression(column: str, values: List[str]) -> str:
     return column + IN + block_expression(values)
 
 
-def regexp_expression(expr: str, pattern: str) -> str:
-    return expr + REG_EXP + pattern
+def regexp_expression(expr: str, pattern: str, parameters: Optional[str] = None) -> str:
+    if parameters is not None:
+        return function_expression("RLIKE", [expr, pattern, parameters], False)
```

Comment:
I'm wondering if this is safe to consolidate `RLIKE` and `REG_EXP`, though they seem be aliases, maybe we can consult the SQL team.

Reply:
`RLIKE` seems to be a super set of functionality for `REG_EXP`. `REG_EXP` does not accept parameters, but a user perspective I think it would be an odd experience to be able to use them with one and not the other since we treat them as aliases in the snowpark api.

Reply:
yeah.. I will accept the fact as we already alias them.